util/codec/codec.go

```diff
@@ -1294,3 +1296,57 @@ func ConvertByCollationStr(str string, tp *types.FieldType) string {
 	collator := collate.GetCollator(tp.Collate)
 	return string(hack.String(collator.Key(str)))
 }
+
+// HashCode encodes a Datum into a unique byte slice.
+// It is mostly the same as EncodeValue, but it doesn't contain truncation or verification logic in order
+// 	to make the encoding lossless.
+func HashCode(b []byte, d types.Datum) []byte {
+	switch d.Kind() {
+	case types.KindInt64:
+		b = encodeSignedInt(b, d.GetInt64(), false)
+	case types.KindUint64:
+		b = encodeUnsignedInt(b, d.GetUint64(), false)
+	case types.KindFloat32, types.KindFloat64:
+		b = append(b, floatFlag)
+		b = EncodeFloat(b, d.GetFloat64())
+	case types.KindString:
+		b = encodeString(b, d, false)
+	case types.KindBytes:
+		b = encodeBytes(b, d.GetBytes(), false)
+	case types.KindMysqlTime:
+		b = append(b, uintFlag)
+		t := d.GetMysqlTime().CoreTime()
```

Comment:
need timezone here?

Reply:
seems ok if hashcode only used to deduplicate expr in one plan

Reply:
Yes. It's only used in the planner and it's in the same session. I think it's OK.